langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiStreamingChatModel.java

```diff
@@ -267,6 +286,51 @@ private void generate(List<ChatMessage> messages,
 
         AzureOpenAiStreamingResponseBuilder responseBuilder = new AzureOpenAiStreamingResponseBuilder(inputTokenCount);
 
+        // Sync version
+        if(client != null) {
+            syncCall(toolThatMustBeExecuted, handler, options, responseBuilder);
+        } else if(asyncClient != null) {
+            asyncCall(toolThatMustBeExecuted, handler, options, responseBuilder);
+        } else {
+            throw new IllegalStateException("No client available");
+        }
+    }
+
+    private void handleResponseException(Exception exception, StreamingResponseHandler<AiMessage> handler) {
+        if (exception instanceof HttpResponseException) {
+            HttpResponseException httpResponseException = (HttpResponseException) exception;
+            logger.info("Error generating response, {}", httpResponseException.getValue());
+            FinishReason exceptionFinishReason = contentFilterManagement(httpResponseException, "content_filter");
+            Response<AiMessage> response =  Response.from(
+                    aiMessage(httpResponseException.getMessage()),
+                    null,
+                    exceptionFinishReason
+            );
+            handler.onComplete(response);
+        } else {
+            handler.onError(exception);
+        }
+    }
+
+    private void asyncCall(ToolSpecification toolThatMustBeExecuted, StreamingResponseHandler<AiMessage> handler, ChatCompletionsOptions options, AzureOpenAiStreamingResponseBuilder responseBuilder) {
+        Flux<ChatCompletions> chatCompletionsStream = asyncClient.getChatCompletionsStream(deploymentName, options);
+
+        chatCompletionsStream.subscribe(chatCompletion -> {
+                    try {
+                        responseBuilder.append(chatCompletion);
+                        handle(chatCompletion, handler);
+                    } catch (Exception exception) {
+                        handleResponseException(exception, handler);
```

Comment:
I guess this handling should be not here but in the error handler below?

Reply:
I think it depends. The one is catching processing of individual response messages and the other one is excepton handler for generating the stream. So, if we remove it, then any exceptions that occur while processing individual items in the stream will stop. The error  propagated to the onError handler of subscribe method. But it should still be fine though. I just propagated to the on errorHandler now.